### DISCUSSION TITLE
Exposing some of the functions to Blueprints

### DISCUSSION THREAD

**Discussion by emrehan98** [Sixze/ALS-Refactored#9] [Ideas]
Hey there, first of all, thank you for your amazing work! I look at the files and everything seems so in order. I'm not a cpp programmer but just started learning it after spending 6 years in Blueprints. I can now do simple things like exposing functions to blueprint but wondering if you'd be open to the idea of exposing some of the functions into the blueprints. If you are open that kind of things, I'd like to list some of them that can be really useful for blueprint programmers. Even the current state of the plugin is pretty usable for someone who hasn't got any cpp experience, but it would be much better if we had more access in blueprints.

But I can definitely understand if my suggestion is simply out of scope. This repository seems pretty refined, made for cpp users.

**Comment by Sixze** [comment]
Hi! There is no problem with exposing some useful functions to blueprints. List which ones you need and I'll see what can be done.

**Reply by emrehan98** [reply]
Thank you! I'm currently diving into your work and trying to figure which functions might be useful. I was thinking about "SetLocomotionMode" first, in order to switch to Swimming/Climbing and other locomotions. So I exposed it by myself but then I realized even though I have "SetLocomotionMode" in blueprints, I'd still need to add/modify couple of things in cpp. (I saw MovementComponent is used for locomotion mode, which makes perfectly sense but it also makes having the BP node of "SetLocomotionMode" kinda useless. So giving suggestions getting harder as I dive into your amazing work. But hopefully I can find some functions and let you know. :)

**Reply by Sixze** [reply]
It's really hard to figure out how to make the plugin easier to use from blueprints.
Unfortunately, this plugin is more of an example of how you can make a character locomotion, and in order to use it in a real project, you still have to understand its internals well and be able to modify them.
And I do not think that with the original blueprint version of the plugin it will be easier, but rather even more difficult, since blueprints are poorly suited for creating complex logic.

**Reply by emrehan98** [reply]
Can't argue with that, as I dive deeper it becomes more clear that I must adapt myself to the plugin rather than trying to make it work with blueprints. Only thing that I think might be useful is "MovementSettings" variable. (DataAsset) It can be used for switching movement speed runtime. (Player gets injured and movement speed changes etc.)

I couldn't find anything except that. So I withdraw my suggestion. By the way, is Discussions section OK to ask questions? 

Thanks for answering. 👍

**Reply by Sixze** [reply]
Yes, I can allow to change the movement settings at runtime, just need to make sure everything still works well in multiplayer.

For questions, there is a discussion category called "Q&A".
